### PR TITLE
interface: react to deleted comments

### LINKED ISSUE #4272: deleting a comment on a link or note gives no feedback
**Describe the bug**
If you hit delete on your comment, nothing seems to happen. It does in fact delete it, but you have to navigate away and back again to see that it's gone

**To Reproduce**
1. Comment on link or note in collection or notebook
2. click delete
3. observe nothing seems to have happened
4. reopen comments and see it has in fact been deleted

**Expected behavior**
Comment should immediately disappear and also probably flash "deleted" or whatever

**Desktop (please complete the following information):**
 - OS: arch linux
 - Browser qutebrowser
 - Base hash eljue

### PR DESCRIPTION
**Pull request by tylershuster**
fixes #4272

@liam-fitzgerald this works, is it sane?

### CODE REVIEW

File: pkg/interface/src/views/components/CommentItem.tsx [hunk 1]
```diff
@@ -18,26 +16,26 @@ const ClickBox = styled(Box)`
...
-  const [revNum, post] = getLatestCommentRevision(comment);
+  const { ship, contacts, name, api, post, index, group } = props;
   const disabled = props.pending || window.ship !== post?.author;
 
-  const onDelete = async () => {
-    await api.graph.removeNodes(ship, name, [comment.post?.index]);
-  };
+  const onDelete = useCallback(async () => {
+    await api.graph.removeNodes(ship, name, [index]);
+  }, [post])
```
Comment (liam-fitzgerald) [inline pkg/interface/src/views/components/CommentItem.tsx:36]: dependencies are wrong

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
minor dependency nitpick, else LGTM!

File: pkg/interface/src/views/components/CommentItem.tsx (hunk 1 above)
Comment (tylershuster) [reply pkg/interface/src/views/components/CommentItem.tsx:36]: D'oh

**Comment by tylershuster** [comment]
Alright, I cleaned that up and the commit history

**Comment by matildepark** [comment]
Honestly in my testing this seems just as spotty as the original behaviour — sometimes a delete just stays there until you navigate and come back; most of the time it disappears fairly quickly. On devnet comments do in fact disappear on a delete but not consistently, sometimes they hang around.

**Pull request closed without merging**